lib/rules/method.js

```diff
@@ -99,15 +99,23 @@ function checkCallExpression(ruleHelper, callExpr, node) {
         break;
     }
 
-    case "AssignmentExpression":
-        if (node.right.type === "MemberExpression") {
-            const newCallExpr = Object.assign({}, callExpr);
-            newCallExpr.callee = node.right;
-            checkCallExpression(ruleHelper, newCallExpr, node.right);
-            break;
-        }
-        checkCallExpression(ruleHelper, callExpr, node.right);
+    case "AssignmentExpression": {
+        // TODO This assumes `node.operator` is always `=`, so `AssignmentExpression`
+        // should evaluate to the right side. This is not true for, for example:
+        // ```
+        // const a = () => console.log('a');
+        // const b = () => console.log('b');
+        // let c = a;
+        // (c ||= b)()
+        // ```
+        // which will print `a`.
```

Comment:
This in itself is a great finding (I somehow did not understand that comment at first and am now just coming back to it).

We can't test whether a statement like `(c ||= b)()` will end up calling `c` or `b`. I suppose we'd have to special-case the handling for different operators.

For the `||=` operator, I suppose we have to follow both? 🤔

Reply:
I shall file a new one for `||=` and such.